src/test/java/fi/helsinki/cs/tmc/core/testhelpers/ProjectRootFinderStub.java

```diff
@@ -33,21 +33,21 @@ private void fillCourseStubs() {
         }
     }
 
-    public String getReturnValue() {
+    public Path getReturnValue() {
         return returnValue;
     }
 
-    public void setReturnValue(String returnValue) {
+    public void setReturnValue(Path returnValue) {
         this.returnValue = returnValue;
     }
 
     @Override
     public Optional<Path> getRootDirectory(Path zipRoot) {
-        return Optional.of(Paths.get(returnValue));
+        return Optional.of(returnValue);
     }
 
-    public Optional<Course> getCurrentCourse(String path) {
-        String[] folders = path.split("\\" + File.separator);
+    public Optional<Course> getCurrentCourse(Path path) {
+        String[] folders = path.toString().split("\\" + File.separator);
```

Comment:
Can we use [`java.nio.Path#iterator()`](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#iterator()) here?